core/src/main/java/smile/regression/OLS.java

```diff
@@ -252,17 +285,26 @@ public OLS(double[][] x, double[] y, boolean SVD) {
                     coefficients[i][1] = se;
                     double t = w1[i] / se;
                     coefficients[i][2] = t;
-                    coefficients[i][3] = Beta.regularizedIncompleteBetaFunction(0.5 * df, 0.5, df / (df + t * t));
+                    coefficients[i][3] = Beta.regularizedIncompleteBetaFunction(0.5 * df, 0.5, df / (df + t * t)); 
                 } else {
                     coefficients[i][1] = Double.NaN;
                     coefficients[i][2] = 0.0;
                     coefficients[i][3] = 1.0;
                 }
+                if (!Math.isZero(s, .01)){
+                    sigmaInverse.set(i, i, 1/Math.sqr(s));
+                }else{
+                    sigmaInverse.set(i, i, 1);
+                }
             }
+            // Calculate pseudoinverse of X^T * X
+            DenseMatrix V = svd.getV();
+            onlineParameters.gamma = V.abmm(sigmaInverse.abmm(V.transpose()));
```

Comment:
don't do transpose V.transpose(). just call sigmaInverse.abtmm(V). It is more efficient.